lib/process_helpers.py

```diff
@@ -2,36 +2,26 @@
 import os
 import subprocess
 
-def kill_ps(ps_to_kill):
-    print('Killing processes')
-    for ps_info in ps_to_kill:
-        pid = ps_info['ps'].pid
-        print(f"Trying to kill {ps_info['cmd']} with PID: {pid}")
-        try:
-            if ps_info['ps_group'] is True:
-                try:
-                    ps_group_id = os.getpgid(pid)
-                    print(f" with process group {ps_group_id}")
-                    os.killpg(os.getpgid(pid), signal.SIGTERM)
-                    try:
-                        ps_info['ps'].wait(timeout=5)
-                    except subprocess.TimeoutExpired:
-                        # If the process hasn't gracefully exited after 5 seconds we kill it
-                        os.killpg(os.getpgid(pid), signal.SIGKILL)
-                except ProcessLookupError:
-                    # process may be not have been in a process group
-                    print(f"Could not find process-group for {pid}")
-            # always, just in case the calling process (typically the shell) did not die
-            os.kill(pid, signal.SIGTERM)
-            try:
-                ps_info['ps'].wait(timeout=5)
-            except subprocess.TimeoutExpired:
-                # If the process hasn't gracefully exited after 5 seconds we kill it
-                os.kill(pid, signal.SIGKILL)
+def kill_pg(ps, pgid, cmd):
+    print(f"Trying to kill {cmd} with PGID: {pgid}")
+
+    os.killpg(pgid, signal.SIGTERM)
+    try:
+        ps.wait(timeout=10)
+    except subprocess.TimeoutExpired as exc:
+        # If the process hasn't gracefully exited after 5 seconds we kill it
+        os.killpg(pgid, signal.SIGKILL)
+        raise RuntimeError(f"Killed the process {cmd} with SIGKILL. This could lead to corrupted data!") from exc
 
-        except ProcessLookupError:
-            # process may already have ended or been killed in the process group
-            print(f"Could not find process {pid}")
+def kill_ps(ps, pid, cmd):
+    print(f"Trying to kill {cmd} with PID: {pid}")
+
+    os.kill(pid, signal.SIGTERM)
+    try:
+        ps.wait(timeout=10)
+    except subprocess.TimeoutExpired as exc:
+        os.kill(pid, signal.SIGKILL)
+        raise RuntimeError(f"Killed the process {cmd} with SIGKILL. This could lead to corrupted data!") from exc
```

Comment:
Couldn't you merge most of the code in `kill_pg` and `kill_ps`. The only thing that is really different is the `os.kill..`

Reply:
yes, you can. But for a ten line function that would need two switch statements to merge I thought it is cleaner to have two separate functions in the end